README.md

```diff
@@ -22,10 +22,27 @@ composer require wyrihaximus/html-compress
 
 require dirname(__DIR__) . '/vendor/autoload.php';
 
-$parser = \WyriHaximus\HtmlCompress\Factory::construct();
+$parser = Factory::constructSmallest();
 $compressedHtml = $parser->compress($sourceHtml);
 ```
 
+## Advanced Usage ##
+
+All factory methods come with a `WithHtmlMin` variant allowing you to set options on the underlying `HtmlMin` instance:
+
+```php
+<?php
+
+require dirname(__DIR__) . '/vendor/autoload.php';
+
+$htmlMin = new HtmlMin();
+$htmlMin->doRemoveHttpPrefixFromAttributes();
+$htmlMin->doMakeSameDomainsLinksRelative(['example.com']);
+
+$parser = Factory::constructSmallestWithHtmlMin($$htmlMin);
+$compressedHtml = $parser->compress($sourceHtml);
+``` 
+
```

Comment:
@abuyoyo What do you think about this?

Reply:
oh yeah. Just saw this. This is a very nice change.
Personally I'm not so happy with the new variant methods required: `constructSmallestWithHtmlMin`.
These can explode exponentially with each new variant added. I personally prefer chain-able instances.

So usage like `Factory::constructSmallest()->setHtmlMin($htmlMin)` or `Factory::setHtmlMin($htmlMin)->constructSmallest()` (which would provide the same result as `constructSmallestWithHtmlMin`)

I think this approach would be more future-proof.

Reply:
But I'm all for the ability to import an `HtmlMin` instance into the factory, instead of just getting access to the default instance.

Reply:
@abuyoyo good point, updated it to be immutable 👍